client/src/store/stores/main.ts

```diff
@@ -212,13 +227,30 @@ export class Store {
       },
     );
 
+    reaction(
```

Comment:
See comment above, we should avoid such reactions which makes top-level logic outside of main control logic components (primarily in `App`). This is also not about being explicit.